### PR TITLE
NMS-16983 : File name field in System Reports is not working.

### PR DESCRIPTION
**Pull request by MusaidAli**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-16983

### CODE REVIEW

**Review by smunir-onms — APPROVED** [review]
LGTM

**Review by smunir-onms — CHANGES_REQUESTED** [review]
Add test, as per instructions by @cgorantla

File: opennms-webapp/pom.xml [hunk 1]
```diff
@@ -985,6 +985,12 @@
       <artifactId>awaitility</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (cgorantla) [inline opennms-webapp/pom.xml:988]: any need for this dependency ?

File: smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package org.opennms.smoketest;
```
Comment (cgorantla) [inline smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java:1]: nit: add missing header

File: smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private class Page {
+        private final String url;
+
+        public Page(String baseUrl) {
+            this.url = Objects.requireNonNull(baseUrl) + "opennms/admin/support/systemReport.htm";
+        }
+        public Page open() {
+            driver.get(url);
+            System.out.println(url);
+            new WebDriverWait(driver, Duration.ofSeconds(10)).until(pageContainsText("Choose which plugins to enable:"));
```
Comment (cgorantla) [inline smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java:48]: I don't find this `Choose which plugins to enable:` on `opennms/admin/support/systemReport.htm`

Comment (MusaidAli) [reply smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java:69]: Screenshot from 2024-12-19 14-44-59

It is before checkboxes ,In first line.

File: smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java (hunk 2 above)
Comment (MusaidAli) [reply smoke-test/src/test/java/org/opennms/smoketest/SystemReportPageIT.java:22]: added header.

**Review by smunir-onms — APPROVED** [review]

**Pull request merged**